packages/utils/telemetry-utils/src/errorLogging.ts

```diff
@@ -3,102 +3,152 @@
-};
-
-/** Inspect the given error for common "safe" props and return them */
-export function extractLogSafeErrorProperties(error: any) {
-    const removeMessageFromStack = (stack: string, errorName?: string) => {
-        const stackFrames = stack.split("\n");
-        stackFrames.shift(); // Remove "[ErrorName]: [ErrorMessage]"
-        if (errorName !== undefined) {
-            stackFrames.unshift(errorName); // Add "[ErrorName]"
-        }
-        return stackFrames.join("\n");
-    };
-
-    const message = (typeof error?.message === "string")
-        ? error.message as string
-        : String(error);
-
-    const safeProps: { message: string; errorType?: string; stack?: string } = {
+/**
+ * Take an unknown error object and extract certain known properties to be included in a new error object.
+ * The stack is preserved, along with any safe-to-log telemetry props.
+ * @param error - An error that was presumably caught, thrown from unknown origins
+ * @param newErrorFn - callback that will create a new error given the original error's message
+ * @returns A new error object "wrapping" the given error
+ */
+ export function wrapError<T>(
+    error: any,
+    newErrorFn: (m: string) => T,
+): T {
+    const {
         message,
-    };
+        stack,
+    } = extractLogSafeErrorProperties(error);
+    const props = isILoggingError(error) ? error.getTelemetryProperties() : {};
 
-    if (isRegularObject(error)) {
-        const { errorType, stack, name } = error;
-
-        if (typeof errorType === "string") {
-            safeProps.errorType = errorType;
-        }
+    const newError = newErrorFn(message);
+    mixinTelemetryProps(newError, props);
 
-        if (typeof stack === "string") {
-            const errorName = (typeof name === "string") ? name : undefined;
-            safeProps.stack = removeMessageFromStack(stack, errorName);
-        }
+    if (stack !== undefined) {
+        Object.assign(newError, { stack });
     }
 
-    return safeProps;
+    return newError;
 }
 
-/** Type guard for ILoggingError */
-export const isILoggingError = (x: any): x is ILoggingError => typeof x?.getTelemetryProperties === "function";
-
 /**
- * Read-Write Logging Error.  Not exported.
- * This type alias includes addTelemetryProperties, and applies to objects even if not instanceof LoggingError\
+ * Normalize the given error object yielding a valid Fluid Error
+ * @returns The same error object passed in if possible, normalized and with any provided annotations applied
+ * @param error - The error to normalize, ideally by patching in the properties of IFluidErrorBase
+ * @param annotations - Annotations to apply to the normalized error:
+ * annotations.props - telemetry props to log with the error
+ * annotations.errorCodeIfNone - fluidErrorCode to mention if error isn't already an IFluidErrorBase
+ * @param strict - If true, then throw if the given error can't be patched. Otherwise, create and return a new object
```

Comment:
I think instead I'll export two functions:

- `normalizeUnknownError` which will be like `strict: false`
- `normalizeErrorObject` which will be like `strict: true`

Thoughts?

Reply:
Or... `normalizeError` and `annotateError`

Reply:
Yeah I went with that.  I like it, `normalizeError` returns `IFluidErrorBase` and `annotateError` asserts that the input is `IFluidErrorBase`, throwing if it can't annotate it.